.vscode/settings.json

```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
+}
```

Comment:
vscode overrides your configurations with this file, it does not replaces the entire settings.json, this is necessary because the tsconfig contains things that are not enabled by default in the global vscode's typescript version, so you will see errors in the editor and those errors are not real because of the misconfiguration of the tsconfig.json, this is a common practice. It also ensures you vscode is using your version of typescript in case you are using a feature that is not available